src/types/shell/xdg_shell_v6.rs

```diff
@@ -181,6 +183,20 @@ impl XdgV6ShellSurface {
         }
     }
 
+    pub fn for_each_surface(&self, mut iterator: &FnMut(SurfaceHandle, i32, i32)) {
+        unsafe {
+            extern fn c_iterator(wlr_surface: *mut wlr_surface, sx: i32, sy: i32, data: *mut c_void) {
+                unsafe {
+                    let iterator = &mut *(data as *mut &mut FnMut(SurfaceHandle, i32, i32));
```

Comment:
You are casting an `*mut & FnMut(SurfaceHandle, i32, i32)` into an *mut &mut FnMut(SurfaceHandle, i32, i32). You should make it take an `&mut FnMut`, or cast this line into an `&FnMut(...)`.